packages/main/src/ColorPalette.ts

```diff
@@ -241,28 +244,75 @@ class ColorPalette extends UI5Element {
 		});
 	}
 
-	_onclick(e: MouseEvent) {
-		const target = e.target as ColorPaletteItem;
-		if (target.hasAttribute("ui5-color-palette-item")) {
-			this.selectColor(target);
+	/**
+	 * In case multiple 'ui5-color-palette-item' elements are explicitly set to be selected,
+	 * this function ensures that only the last one remains active, preventing multiple selections.
+	 *
+	 * @private
+	 * @returns {void}
+	 */
+	_filterSelectedItems() {
+		const allColorPaletteItems = this.colors;
+
+		const selectedItems = allColorPaletteItems.filter(item => item.selected);
+		if (selectedItems.length > 1) {
+			for (let i = 0; i < selectedItems.length - 1; i++) {
+				(selectedItems[i]).selected = !(selectedItems[i]).selected;
+			}
 		}
 	}
 
+	/**
+	 * Deselects all items in the color palette.
+	 *
+	 * @private
+	 * @returns {void}
+	 */
+	_deselectAllItems(items: ColorPaletteItem[]) {
```

Comment:
You can think on combining this method with previous one, as _deselectAllItems(items: ColorPaletteItem[]) deselects all items and _filterSelectedItems() deselects all items without the last one